DDG4/CMakeLists.txt

```diff
@@ -104,12 +104,19 @@ ENDIF()
 
 IF(TARGET EDM4HEP::edm4hep)
   dd4hep_add_plugin(DDG4EDM4HEP
-    SOURCES edm4hep/*.cpp
+    SOURCES edm4hep/Geant4Output2EDM4hep.cpp
     USES    DD4hep::DDG4 EDM4HEP::edm4hep EDM4HEP::edm4hepDict podio::podio podio::podioDict podio::podioRootIO
     )
   install(TARGETS DDG4EDM4HEP EXPORT DD4hep LIBRARY DESTINATION lib)
   set_target_properties(DDG4EDM4HEP PROPERTIES VERSION ${DD4hep_VERSION} SOVERSION ${DD4hep_SOVERSION})
-
+  if(TARGET(podio::podioIO))
+      dd4hep_add_plugin(DDG4EDM4HEPReader
+        SOURCES edm4hep/EDM4hepFileReader.cpp
+        USES    DD4hep::DDG4 EDM4HEP::edm4hep EDM4HEP::edm4hepDict podio::podio podio::podioDict podio::podioIO
+      )
+      install(TARGETS DDG4EDM4HEPReader EXPORT DD4hep LIBRARY DESTINATION lib)
+      set_target_properties(DDG4EDM4HEPReader PROPERTIES VERSION ${DD4hep_VERSION} SOVERSION ${DD4hep_SOVERSION})
+    endif()
```

Comment:
I think a warning or at least a status message would be nice here too, to explain why the EDM4hep input reader is not built. That might avoid some confusion and could also be used to provide the solution along side it.

Reply:
Thanks! Also the IF check was wrong.